unified_planning/model/action.py

```diff
@@ -485,11 +486,9 @@ def add_decrease_effect(self, timing: 'up.model.timing.Timing', fluent: Union['u
                                          condition_exp, kind = up.model.effect.EffectKind.DECREASE))
 
     def _add_effect_instance(self, timing: 'up.model.timing.Timing', effect: 'up.model.effect.Effect'):
-        if timing in self._effects:
-            if effect not in self._effects[timing]:
-                self._effects[timing].append(effect)
-        else:
-            self._effects[timing] = [effect]
+        concurrent_effects = self._effects.setdefault(timing, [])
```

Comment:
Why do you need to store a list of effects? A **set** of fluent expressions associated to a timing should be enough (and way more efficient!)